tests/mainTests.cpp

```diff
@@ -213,10 +220,10 @@ void copyingTests()
     assertEquals(testList.size(), copiedList.size());
     for (int i=0; i < testList.size(); i++)
         assertEquals(testList.get(i), copiedList.get(i));
-    testListIter = testList.iterator();
-    copiedListIter = copiedList.iterator();
-    while (testListIter->hasNext())
-        assertEquals(testListIter->next(), copiedListIter->next());
+    testListIter.reset(testList);
+    copiedListIter.reset(copiedList);
+    while (testListIter.hasNext())
+        assertEquals(testListIter.next(), copiedListIter.next());
```

Comment:
After this loop add assertions that check if both iterators hasNext() methods return false